helusers/jwt.py

```diff
@@ -24,13 +28,24 @@ def validate(self, keys, audience):
         and validates the claims, raising an exception if anything fails."""
 
         options = {
-            "require_aud": True,
+            "verify_aud": False,
             "require_exp": True,
         }
 
-        jwt.decode(
-            self._encoded_jwt, keys, options=options, audience=audience
-        )
+        jwt.decode(self._encoded_jwt, keys, options=options)
+
+        claims = self.claims
+
+        if "aud" not in claims:
+            raise ValidationError("Missing required 'aud' claim.")
+
+        claim_audiences = claims["aud"]
```

Comment:
Is it guaranteed that claim_audiences is eihter a str or set here?

Reply:
Probably not. It doesn't have to be a `set` though, it's enough that it's an iterable (so `list` is fine). But I guess it could also be a boolean, a number or `None` too -- anything that can come out of JSON decoding.

Reply:
With a boolean for example, this wouldn't raise a ValidationError, but maybe ValueError or something. Maybe if we want to systematically see ValidationErrors from logs that could be a problem?

Reply:
The `oidc.RequestJWTAuthentication` (where this is called from) turns any `Exception` thrown here into `oidc.AuthenticationError`, so that would probably end up into logs etc.